### PR TITLE
Avoid unnecessary use of Linq

### PR DESCRIPTION
**Pull request by jkotas**

### CODE REVIEW

File: src/libraries/Microsoft.Extensions.DependencyInjection/tests/Utils/MultiServiceHelpers.cs [hunk 1]
```diff
@@ -43,7 +42,7 @@ private static bool IsGenericIEnumerable(Type type)
 
         private static Type FirstGenericArgument(Type type)
         {
-            return type.GetTypeInfo().GenericTypeArguments.Single();
+            return type.GetTypeInfo().GenericTypeArguments[0];
```
Comment (GrabYourPitchforks) [inline src/libraries/Microsoft.Extensions.DependencyInjection/tests/Utils/MultiServiceHelpers.cs]: Isn't this a behavioral change? Seems like it's the equivalent of `First()`, not `Single()`.

Comment (GrabYourPitchforks) [reply src/libraries/Microsoft.Extensions.DependencyInjection/tests/Utils/MultiServiceHelpers.cs]: Ah, this code path is only used for `IEnumerable<>` cases. Carry on. :)

**Review by GrabYourPitchforks — APPROVED** [review]

**Comment by VSadov** [comment]
I wonder how this was found? Came up in some profiles or just by noticing `using System.Linq`?

My personal favorite is `Count` vs. `Count()` vs. `Length` - one is not like the others, but easy to miss and sometimes shows up in really unexpected places when profiled.

**Review by VSadov — APPROVED** [review]

**Comment by jkotas** [comment]
> I wonder how this was found? 

I noticed this when debugging unrelated crash.

**Comment by davidfowl** [comment]
Nice!

**Pull request merged**